classes/Provider/TalkEmailerServiceProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Provider;
+
+use OpenCFP\Domain\Services\TalkEmailer;
+use Pimple\Container;
+use Pimple\ServiceProviderInterface;
+use Swift_Mailer;
+use Swift_SmtpTransport;
+
+final class TalkEmailerServiceProvider implements ServiceProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function register(Container $app)
+    {
+        $options = [
+            'host'     => $app->config('mail.host'),
+            'port'     => $app->config('mail.port'),
+            'username' => $app->config('mail.username'),
+            'password' => $app->config('mail.password'),
+        ];
+        $transport = (new Swift_SmtpTransport($options['host'], $options['port']))
+            ->setUsername($options['username'])
+            ->setPassword($options['password']);
+        $swiftMailer         = new Swift_Mailer($transport);
```

Comment:
As of the moment, we create all of these even if we never use the service.

How about moving all of this into the closure, and thus simplifying this method to

```php
public function register(Container $app)
{
    $app['talk_emailer'] = function ($app) {
        /** @var Application $app */
        $transport = new Swift_SmtpTransport(
            $app->config('mail.host'),
            $app->config('mail.port')
        );

        $transport
            ->setUsername($app->config('mail.username'))
            ->setPassword($app->config('mail.password'));

        return new TalkEmailer(new Swift_Mailer($transport));
    };
}
```

?